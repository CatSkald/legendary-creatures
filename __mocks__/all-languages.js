import languages from "../src/i18n/languages";

exports.allLanguages = Object.entries(languages).map(
  ([lang, langProps]) => langProps,
);
