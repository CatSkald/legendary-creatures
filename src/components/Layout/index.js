import "../../styles/global.scss";
import "../../styles/theme-light.scss";
import "../../styles/theme-dark.scss";
import styles from "./index.module.scss";

import React, { useState } from "react";
import Header from "../Header";
import Footer from "../Footer";

const LocaleContext = React.createContext();

const Layout = ({ children, pageContext: { language } }) => {
  const [isDarkColorTheme, switchColorTheme] = useState(false);

  function handleToggleColorTheme() {
    switchColorTheme(!isDarkColorTheme);
  }

  return (
    <LocaleContext.Provider value={{ language }}>
      <div
        className={`body ${styles.Wrapper} ${
          isDarkColorTheme ? "theme-dark" : "theme-light"
        }`}
      >
        <Header
          isDarkColorTheme={isDarkColorTheme}
          handleToggleColorTheme={handleToggleColorTheme}
        />
        <section className={styles.SiteContent} role="main">
          <div className={styles.Container}>{children}</div>
        </section>
        <Footer />
      </div>
    </LocaleContext.Provider>
  );
};

export { Layout, LocaleContext };
