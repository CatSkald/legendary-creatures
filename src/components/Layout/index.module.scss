.layout__container {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.layout {
  display: flex;
  flex: 1;
  width: 100%;
}

.layout__content {
  width: 100%;
  max-width: var(--container-width);
  margin: 0 auto;
  padding: var(--space-sm);

  @media screen and (min-width: 768px /*--screen-medium*/) {
    padding: var(--space);
  }
}
