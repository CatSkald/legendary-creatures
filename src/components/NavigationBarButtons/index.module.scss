.active {
  /* placeholder to be referenced from styles */
}

.ButtonList {
  display: flex;
  right: var(--space-sm);

  @media screen and (min-width: 768px /*--screen-medium*/) {
    margin-left: 0;
    position: relative;
    right: 0;
  }
}

.Button {
  margin-left: var(--space-sm);
  display: inline-block;
  color: var(--menu-text);
  text-decoration: none;
  min-width: 16px;
  cursor: pointer;

  &:hover {
    color: var(--menu-hover);
  }

  @media screen and (max-width: 768px /*--screen-medium*/) {
    font-size: 18px;
  }
}

.Language {
  &:after {
    content: " ⤋"; //non-breaking space
  }

  &.active:after {
    content: " ⤊"; //non-breaking space
  }
}

.ColorTheme {
  font-weight: bold;
}
