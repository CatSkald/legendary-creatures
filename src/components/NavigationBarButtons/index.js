import {
  active,
  navigationBarButtons,
  navigationBarButtons__button,
  navigationBarButtons__language,
  navigationBarButtons__theme,
  navigationBarButtons__theme__dark,
  navigationBarButtons__theme__light,
} from "./index.module.scss";

import React from "react";
import PropTypes from "prop-types";
import { LocaleContext } from "../Layout";

const NavigationBarButtons = (props) => {
  const { language } = React.useContext(LocaleContext);

  return (
    <ul className={navigationBarButtons}>
      <li className={navigationBarButtons__button}>
        <button
          title="Change language"
          className={`${navigationBarButtons__language} ${
            props.isLanguageSelectionActive ? active : ""
          }`}
          onClick={() => props.handleToggleLanguageSelection()}
        >
          {language.name}
        </button>
      </li>
      <li className={navigationBarButtons__button}>
        <button
          title="Switch color theme"
          aria-label="Change website color theme"
          onClick={() => props.handleToggleColorTheme()}
        >
          <span
            role="img"
            className={`${navigationBarButtons__theme} ${
              props.isDarkColorTheme
                ? navigationBarButtons__theme__dark
                : navigationBarButtons__theme__light
            }`}
          ></span>
        </button>
      </li>
    </ul>
  );
};

NavigationBarButtons.propTypes = {
  handleToggleColorTheme: PropTypes.func.isRequired,
  isDarkColorTheme: PropTypes.bool.isRequired,
  handleToggleLanguageSelection: PropTypes.func.isRequired,
  isLanguageSelectionActive: PropTypes.bool.isRequired,
};

export default NavigationBarButtons;
