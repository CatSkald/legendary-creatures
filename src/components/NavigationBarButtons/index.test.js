import React from "react";
import { render } from "@testing-library/react";

import { allLanguages } from "../../../__test-helpers__/all-languages.helper";
import { LocaleContext } from "../Layout";
import NavigationBarButtons from "../NavigationBarButtons";

describe("NavigationBarButtons", () => {
  const mock = jest.fn();
  const theme = Object.freeze({
    name: "Test Theme",
    icon: "⨁",
    cssClassName: "theme-custom",
  });

  beforeEach(() => {
    mock.mockClear();
  });

  describe.each(allLanguages)(
    "renders correctly for language %s",
    (language) => {
      test.each([false, true])(
        "when isLanguageSelectionActive %s",
        (isLanguageSelectionActive) => {
          const { container } = render(
            <LocaleContext.Provider value={{ language }}>
              <NavigationBarButtons
                handleToggleColorTheme={mock}
                handleToggleLanguageSelection={mock}
                selectedTheme={theme}
                isLanguageSelectionActive={isLanguageSelectionActive}
              />
            </LocaleContext.Provider>,
          );
          expect(container).toMatchSnapshot();
        },
      );
    },
  );
});
