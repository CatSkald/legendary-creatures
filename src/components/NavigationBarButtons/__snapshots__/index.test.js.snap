// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBarButtons renders correctly for language {
  name: 'Українська',
  code: 'uk',
  path: '/uk',
  dateFormat: 'DD.MM.YYYY'
} when isDarkColorTheme false and isLanguageSelectionActive false 1`] = `
<div>
  <ul
    class="navigationBarButtons"
  >
    <li
      class="navigationBarButtons__button"
    >
      <button
        class="navigationBarButtons__language "
        title="Change language"
      >
        Українська
      </button>
    </li>
    <li
      class="navigationBarButtons__button"
    >
      <button
        aria-label="Change website color theme"
        title="Switch color theme"
      >
        <span
          class="navigationBarButtons__theme navigationBarButtons__theme__light"
          role="img"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`NavigationBarButtons renders correctly for language {
  name: 'Українська',
  code: 'uk',
  path: '/uk',
  dateFormat: 'DD.MM.YYYY'
} when isDarkColorTheme false and isLanguageSelectionActive true 1`] = `
<div>
  <ul
    class="navigationBarButtons"
  >
    <li
      class="navigationBarButtons__button"
    >
      <button
        class="navigationBarButtons__language active"
        title="Change language"
      >
        Українська
      </button>
    </li>
    <li
      class="navigationBarButtons__button"
    >
      <button
        aria-label="Change website color theme"
        title="Switch color theme"
      >
        <span
          class="navigationBarButtons__theme navigationBarButtons__theme__light"
          role="img"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`NavigationBarButtons renders correctly for language {
  name: 'Українська',
  code: 'uk',
  path: '/uk',
  dateFormat: 'DD.MM.YYYY'
} when isDarkColorTheme true and isLanguageSelectionActive false 1`] = `
<div>
  <ul
    class="navigationBarButtons"
  >
    <li
      class="navigationBarButtons__button"
    >
      <button
        class="navigationBarButtons__language "
        title="Change language"
      >
        Українська
      </button>
    </li>
    <li
      class="navigationBarButtons__button"
    >
      <button
        aria-label="Change website color theme"
        title="Switch color theme"
      >
        <span
          class="navigationBarButtons__theme navigationBarButtons__theme__dark"
          role="img"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`NavigationBarButtons renders correctly for language {
  name: 'Українська',
  code: 'uk',
  path: '/uk',
  dateFormat: 'DD.MM.YYYY'
} when isDarkColorTheme true and isLanguageSelectionActive true 1`] = `
<div>
  <ul
    class="navigationBarButtons"
  >
    <li
      class="navigationBarButtons__button"
    >
      <button
        class="navigationBarButtons__language active"
        title="Change language"
      >
        Українська
      </button>
    </li>
    <li
      class="navigationBarButtons__button"
    >
      <button
        aria-label="Change website color theme"
        title="Switch color theme"
      >
        <span
          class="navigationBarButtons__theme navigationBarButtons__theme__dark"
          role="img"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`NavigationBarButtons renders correctly for language { name: 'English', code: 'en', path: '', dateFormat: 'DD/MM/YYYY' } when isDarkColorTheme false and isLanguageSelectionActive false 1`] = `
<div>
  <ul
    class="navigationBarButtons"
  >
    <li
      class="navigationBarButtons__button"
    >
      <button
        class="navigationBarButtons__language "
        title="Change language"
      >
        English
      </button>
    </li>
    <li
      class="navigationBarButtons__button"
    >
      <button
        aria-label="Change website color theme"
        title="Switch color theme"
      >
        <span
          class="navigationBarButtons__theme navigationBarButtons__theme__light"
          role="img"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`NavigationBarButtons renders correctly for language { name: 'English', code: 'en', path: '', dateFormat: 'DD/MM/YYYY' } when isDarkColorTheme false and isLanguageSelectionActive true 1`] = `
<div>
  <ul
    class="navigationBarButtons"
  >
    <li
      class="navigationBarButtons__button"
    >
      <button
        class="navigationBarButtons__language active"
        title="Change language"
      >
        English
      </button>
    </li>
    <li
      class="navigationBarButtons__button"
    >
      <button
        aria-label="Change website color theme"
        title="Switch color theme"
      >
        <span
          class="navigationBarButtons__theme navigationBarButtons__theme__light"
          role="img"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`NavigationBarButtons renders correctly for language { name: 'English', code: 'en', path: '', dateFormat: 'DD/MM/YYYY' } when isDarkColorTheme true and isLanguageSelectionActive false 1`] = `
<div>
  <ul
    class="navigationBarButtons"
  >
    <li
      class="navigationBarButtons__button"
    >
      <button
        class="navigationBarButtons__language "
        title="Change language"
      >
        English
      </button>
    </li>
    <li
      class="navigationBarButtons__button"
    >
      <button
        aria-label="Change website color theme"
        title="Switch color theme"
      >
        <span
          class="navigationBarButtons__theme navigationBarButtons__theme__dark"
          role="img"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`NavigationBarButtons renders correctly for language { name: 'English', code: 'en', path: '', dateFormat: 'DD/MM/YYYY' } when isDarkColorTheme true and isLanguageSelectionActive true 1`] = `
<div>
  <ul
    class="navigationBarButtons"
  >
    <li
      class="navigationBarButtons__button"
    >
      <button
        class="navigationBarButtons__language active"
        title="Change language"
      >
        English
      </button>
    </li>
    <li
      class="navigationBarButtons__button"
    >
      <button
        aria-label="Change website color theme"
        title="Switch color theme"
      >
        <span
          class="navigationBarButtons__theme navigationBarButtons__theme__dark"
          role="img"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`NavigationBarButtons renders correctly for language { name: 'Русский', code: 'ru', path: '/ru', dateFormat: 'DD.MM.YYYY' } when isDarkColorTheme false and isLanguageSelectionActive false 1`] = `
<div>
  <ul
    class="navigationBarButtons"
  >
    <li
      class="navigationBarButtons__button"
    >
      <button
        class="navigationBarButtons__language "
        title="Change language"
      >
        Русский
      </button>
    </li>
    <li
      class="navigationBarButtons__button"
    >
      <button
        aria-label="Change website color theme"
        title="Switch color theme"
      >
        <span
          class="navigationBarButtons__theme navigationBarButtons__theme__light"
          role="img"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`NavigationBarButtons renders correctly for language { name: 'Русский', code: 'ru', path: '/ru', dateFormat: 'DD.MM.YYYY' } when isDarkColorTheme false and isLanguageSelectionActive true 1`] = `
<div>
  <ul
    class="navigationBarButtons"
  >
    <li
      class="navigationBarButtons__button"
    >
      <button
        class="navigationBarButtons__language active"
        title="Change language"
      >
        Русский
      </button>
    </li>
    <li
      class="navigationBarButtons__button"
    >
      <button
        aria-label="Change website color theme"
        title="Switch color theme"
      >
        <span
          class="navigationBarButtons__theme navigationBarButtons__theme__light"
          role="img"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`NavigationBarButtons renders correctly for language { name: 'Русский', code: 'ru', path: '/ru', dateFormat: 'DD.MM.YYYY' } when isDarkColorTheme true and isLanguageSelectionActive false 1`] = `
<div>
  <ul
    class="navigationBarButtons"
  >
    <li
      class="navigationBarButtons__button"
    >
      <button
        class="navigationBarButtons__language "
        title="Change language"
      >
        Русский
      </button>
    </li>
    <li
      class="navigationBarButtons__button"
    >
      <button
        aria-label="Change website color theme"
        title="Switch color theme"
      >
        <span
          class="navigationBarButtons__theme navigationBarButtons__theme__dark"
          role="img"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`NavigationBarButtons renders correctly for language { name: 'Русский', code: 'ru', path: '/ru', dateFormat: 'DD.MM.YYYY' } when isDarkColorTheme true and isLanguageSelectionActive true 1`] = `
<div>
  <ul
    class="navigationBarButtons"
  >
    <li
      class="navigationBarButtons__button"
    >
      <button
        class="navigationBarButtons__language active"
        title="Change language"
      >
        Русский
      </button>
    </li>
    <li
      class="navigationBarButtons__button"
    >
      <button
        aria-label="Change website color theme"
        title="Switch color theme"
      >
        <span
          class="navigationBarButtons__theme navigationBarButtons__theme__dark"
          role="img"
        />
      </button>
    </li>
  </ul>
</div>
`;
